### PR TITLE
Simple implementation for `get` and `set` without pydash

### PR DESCRIPTION
**Pull request by minhtuev**
## What changes are proposed in this pull request?

Simple implementation for `get` and `set` in python panel 

## How is this patch tested? If it is not, please explain why.

* Manual testing (✅ )
* Unit tests (TBD)

### Is this a user-facing change that should be mentioned in the release notes?

-   [x] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [x] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
  - Improved control over setting and getting values in the PanelConfig class, allowing for better error handling and structure validation.

- **Bug Fixes**
  - Enhanced clarity and robustness in dictionary manipulation, eliminating issues related to reliance on external libraries.

- **Chores**
  - Removed the dependency on `pydash` from project requirements to streamline dependencies and improve maintainability.

### CODE REVIEW

**Comment by minhtuev** [comment]
@ritch : we were discussing a little bit about how to slim down fiftyone package a little bit :) not sure if I understand all the use cases for the function, but here is a simple stab 🙏

**Review by benjaminpkane — CHANGES_REQUESTED** [review]
Looks like some pydash still exist. Can you also target `release/v0.25.0`?

**Comment by minhtuev** [comment]
@benjaminpkane : I think these pydash calls are in teams. We can move this to a fiftyone util lib like eta once we agree this is right approach :)

File: fiftyone/operators/panel.py [hunk 1]
```diff
@@ -312,3 +310,37 @@ def set_title(self, title):
...
+
+def _get(data, key, default=None):
+    parts = key.split(".")
+    if len(parts) == 1:
+        return data.get(key, default)
+
+    current_dict = data
+    for part in parts[:-1]:
+        if (
+            not isinstance(current_dict, dict)
```
Comment (ritch) [inline fiftyone/operators/panel.py:323]: What if its a list?

**Comment by imanjra** [comment]
I think it might be better for us to borrow get and set impl from pydash lib.

**Review by benjaminpkane — APPROVED** [review]
As to not block the release, let's just remove the `8.0.1` pin but still ship with `pydash`

**Pull request merged**